libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1064,6 +1045,29 @@ string YulUtilFunctions::arrayLengthFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::extractByteArrayLengthFunction()
+{
+	string functionName = "extract_byte_array_length";
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers w(R"(
+			function <functionName>(data) -> length {
+				length := div(data, 2)
+				let outOfPlaceEncoding := and(data, 1)
+				if iszero(outOfPlaceEncoding) {
+					length := and(data, 0x7f)
```

Comment:
Currently this overwrites `length`and the result of division by 2 gets lost.

```suggestion
					length := and(length, 0x7f)
```